mm2src/mm2_bitcoin/crypto/src/lib.rs

```diff
@@ -102,6 +103,8 @@ pub fn checksum(data: &[u8], sum_type: &ChecksumType) -> H32 {
         ChecksumType::DSHA256 => result.copy_from_slice(&dhash256(data)[0..4]),
         ChecksumType::DGROESTL512 => result.copy_from_slice(&dgroestl512(data)[0..4]),
         ChecksumType::KECCAK256 => result.copy_from_slice(&keccak256(data)[0..4]),
+        // TODO: segwit checksum
```

Comment:
TODO

Reply:
I found that implementing this will add a lot of code that does nothing because checksum functions are only used for legacy addresses to display and convert from string and for segwit this is used https://github.com/KomodoPlatform/atomicDEX-API/blob/e97e2fc32315ba54d8bc4adc0ce24f7820d1dbcc/mm2src/mm2_bitcoin/keys/src/address.rs#L200
which checks if the checksum is correct in the decode function
https://github.com/KomodoPlatform/atomicDEX-API/blob/e97e2fc32315ba54d8bc4adc0ce24f7820d1dbcc/mm2src/mm2_bitcoin/keys/src/segwitaddress.rs#L153